app/src/main/java/gholzrib/arctouchchallenge/ui/adapters/MoviesAdapter.java

```diff
@@ -0,0 +1,241 @@
+            backdropSizePath = tmdbConfiguration.getImageSizeUrlPath(IMAGE_TYPE_BACKDROP, imageWidth);
+            //Log.i(TAG, "Poster path: " + posterSizePath + " Backdrop path: " + backdropSizePath);
+        }
+        return new MovieHolder(itemView);
+    }
+
+    @Override
+    public void onBindViewHolder(final MovieHolder holder, int position) {
+        final Movie movie = mUpcomingMoviesList.get(position);
+
+        LinearLayout.LayoutParams params = (LinearLayout.LayoutParams) holder.mRltPosterContainer.getLayoutParams();
+        params.height = imageDimen;
+        holder.mRltPosterContainer.setLayoutParams(params);
+
+        String posterUrl = null;
+        if (movie.getPoster_path() != null) {
+            posterUrl = Constants.BASE_URL_IMAGES + posterSizePath + movie.getPoster_path();
+        } else if (movie.getBackdrop_path() != null) {
+            posterUrl = Constants.BASE_URL_IMAGES + backdropSizePath + movie.getBackdrop_path();
+        }
+
+        if (posterUrl != null && CheckConnection.hasInternetConnection(mContext, false)) {
+            //Log.d(TAG, "Poster URL: " + posterUrl);
+            Picasso.with(mContext)
+                    .load(posterUrl)
+                    .placeholder(R.drawable.placeholder_generic)
+                    .into(holder.mIvPoster, new Callback() {
+                        @Override
+                        public void onSuccess() {
+                            holder.mPrgLoading.setVisibility(View.GONE);
+                        }
+
+                        @Override
+                        public void onError() {
+                            holder.mPrgLoading.setVisibility(View.GONE);
+                        }
+                    });
+        } else {
+            holder.mPrgLoading.setVisibility(View.GONE);
+            holder.mIvPoster.setImageResource(R.drawable.placeholder_generic);
+        }
+
+        holder.mTvTitle.setText(movie.getTitle());
+
+        try {
+            String date = appSdf.format(apiSdf.parse(movie.getRelease_date()));
+            holder.mTvReleaseDate.setText(date);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+
+        holder.mLnrGenres.removeAllViews();
+        holder.mLnrGenres.post(new Runnable() {
+            @Override
+            public void run() {
+                int margin = mContext.getResources().getDimensionPixelSize(R.dimen.spacing_tiny);
+                if (holder.mLnrGenres.getWidth() > 0) {
+                    LNR_GENRES_WIDTH_LIMIT = holder.mLnrGenres.getWidth();
+                }
+
+                LinearLayout mLnrGenresRow = createGenresRowLayout();
+
+                for (int i = 0; i < movie.getGenres().size(); i++) {
+                    Movie.MovieGenre genre = movie.getGenres().get(i);
+                    TextView mTvGenre = new TextView(mContext);
+                    LinearLayout.LayoutParams genreParams = new LinearLayout.LayoutParams(
```

Comment:
I didn't realize that I could just inflate a layout file instead of always create a new one programmatically.  This is the way I usually handle this kind of feature, but I will try a different approach next time.